core/src/main/java/org/opensearch/sql/planner/optimizer/rule/PushPageSize.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.planner.optimizer.rule;
+
+import static org.opensearch.sql.planner.optimizer.pattern.Patterns.pagination;
+
+import com.facebook.presto.matching.Capture;
+import com.facebook.presto.matching.Captures;
+import com.facebook.presto.matching.Pattern;
+import lombok.Getter;
+import lombok.experimental.Accessors;
+import org.opensearch.sql.planner.logical.LogicalPaginate;
+import org.opensearch.sql.planner.logical.LogicalPlan;
+import org.opensearch.sql.planner.logical.LogicalRelation;
+import org.opensearch.sql.planner.optimizer.Rule;
+
+public class PushPageSize
```

Comment:
https://github.com/Bit-Quill/opensearch-project-sql/pull/226#discussion_r1144282079 by @MaxKsyunz:

> Could you implement this similar to `PUSH_DOWN_*` rules:
> * Add a `pushDownPageSize` method to  `TableScanBuilder`
> * Add a `PUSH_DOWN_PAGE_SIZE` similar to others that matches `LogicalPaginate` of `scanBuilder` and calls `pushDownPageSize`.
> 
> Doing so would remove the need for `PushPageSize` and `LogicalRelation.pageSize` property, be less code, and follow a pattern already used in the code base, like `PUSH_DOWN_LIMIT`.

Reply:
I tried to do this and faced #1470. I keep the draft, and I'll try to do thorough investigation and fix.